numpy/lib/type_check.py

```diff
@@ -386,13 +392,14 @@ def nan_to_num(x, copy=True):
     """
     x = _nx.array(x, subok=True, copy=copy)
     xtype = x.dtype.type
+    isscalar = (x.ndim == 0)
+    x = x[None] if isscalar else x
+
     if not issubclass(xtype, _nx.inexact):
-        return x
+        return x[0] if isscalar else x
```

Comment:
No need to cast to 1d, just use `x[()]`

Reply:
Oh, I see this matches what happens below too, so is at least consistent

Reply:
Another option is to simply use `asscalar`. What do you think?

Reply:
Nice catch